### DISCUSSION TITLE
Foundry Scripts Broadcast Gas?

### DISCUSSION THREAD

**Discussion by 0xMySt1c** [foundry-rs/foundry#5304] [Q&A]
If I use a public variable in my script contract in the context of vm.broadcast does it cost extra gas to broadcast?

**Comment by mds1** [comment]
Assuming you mean this:

```solidity
contract MyScript is Script {
  uint256 public myVar = 10; // <-- The public var you are referring to

  function run() public {
    vm.broadcast();
    A.foo(myVar); // <-- The broadcasted transaction
  }
}
```

In that example, it does not matter whether `myVar` is public or internal—the gas cost of the broadcasted transaction is the same

**Reply by 0xMySt1c** [reply]
```solidity
contract MyScript is Script {
  address public a;
  address public b;

  function run() public {
    vm.startBroadcast();
    a = deployCode("Token.sol");
    b = deployCode("Contract.sol", abi.encode(a));
  }
}
```

how about the scenario above? where `a` is set and used in the context of broadcast?

**Reply by 0xMySt1c** [reply]
i guess another question: does setting a public var in broadcast cost gas?
